src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -468,6 +469,14 @@ public override async Task RunAsync()
             var httpOptions = hostService.Services.GetRequiredService<IOptions<HttpOptions>>();
             if (scriptHost != null && scriptHost.Functions.Any())
             {
+                // Checking if in Limelight - it should have a `AzureDevSessionsRemoteHostName` value in local.settings.json.
+                var forwardedHttpUrl = _secretsManager.GetSecrets().FirstOrDefault(
+                    s => s.Key.Equals(Constants.AzureDevSessionsRemoteHostName, StringComparison.OrdinalIgnoreCase)).Value;
+                if (forwardedHttpUrl != null){
+                    Regex rgx = new Regex("<port>");
+                    baseUri = rgx.Replace(forwardedHttpUrl, Port.ToString(), 1);
```

Comment:
Regex is fine. Since it is a simple string replacement, I would recommend using `String.Replace`. I also recommend moving `<port>` placeholder to Constants. 

`baseUri = forwardedHttpUrl.Replace(Constants.PortPlaceholder, Port.ToString(), StringComparison.OrdinalIgnoreCase);`